src/ol/source/ImageWMS.js

```diff
@@ -191,6 +191,43 @@ class ImageWMS extends ImageSource {
       1, sourceProjectionObj || projectionObj, baseParams);
   }
 
+  /**
+   * Return the GetLegendGraphic URL for the passed resolution.
+   * Return `undefined` if the GetLegendGraphic URL cannot be constructed.
+   * @param {!number} resolution Resolution. If set to undefined `SCALE`
```

Comment:
Optional parameters should be using the `{Type} [name]` syntax now, I know this is not consistent with the rest of the file but this is the correct JSDoc syntax nowadays.

See for example this recently merged PR: https://github.com/openlayers/openlayers/pull/9751

Reply:
I'll do this in a follow-up. Thanks for you review.

Reply:
See https://github.com/openlayers/openlayers/pull/9792